src/server.c

```diff
@@ -11,6 +11,9 @@
 #include <wlr/types/wlr_primary_selection_v1.h>
 #include <wlr/types/wlr_screencopy_v1.h>
 #include <wlr/types/wlr_viewporter.h>
+#if HAVE_XWAYLAND
+#include <wlr/xwayland.h>
+#endif
```

Comment:
We could remove the last `<wlr/xwayland.h>` include completely by adding a filter function to `xwayland.c`. Something like
```c
bool
xwayland_allow_global(struct server *server, const struct wl_client *client,
		const struct wl_interface *interface)
{
	struct wl_client *xwayland_client =
		server->xwayland ? server->xwayland->server->client : NULL;
	if (!xwayland_client || client != xwayland_client) {
		return true;
	}

	// Comment from original function
	if (!strcmp(iface->name, wp_drm_lease_device_v1_interface.name)) {
		return false;
	}
	return true;
}
```

and then call it from `server.c` like
```c
	[..]
	// Potentially further checks in the future
#if HAVE_XWAYLAND
	if (!xwayland_allow_global(server, client, iface)) {
		return false;
	}
#endif
	// Potentially further checks in the future
	return true;
```

That would also require to move some more includes to `xwayland.c` though, like `"drm-lease-v1-protocol.h"` and potentially `<wayland-server-core.h>` (not sure about the last one).

Reply:
I wasn't sure what to do with the drm-lease stuff, mainly since I don't really understand what it's doing (some kind of a workaround from what I could see?)

Is it all right to leave that for a future MR?

Reply:
The DRM lease protocol allows applications to exclusively take over the whole output, it is most commonly used for VR headsets but there is also a branch of gamescope which can take advantage of it to render games directly to the output instead of going through labwc.

Sadly the xwayland implementation for leased outputs had (or still has) a bug which causes some applications (mainly chromium based ones running in X11 mode) to lag over time. To prevent that from happening we just stop announcing lease support to xwayland (as its itself a wayland client) until there is a better solution available.

> Is it all right to leave that for a future MR?

Sure.